### PR TITLE
Did some extensive editing of the grammar tutorial.

### PR DESCRIPTION
**Pull request by antquinonez**
Would invite feedback, suggestions for changes.

### CODE REVIEW

**Comment by Altai-man** [comment]
Unfortunately, I cannot properly review it(hence the delay), though the conflicts were cause by my last merge. They need to be resolved before we move forward. I'll cast someone great enough to review it tomorrow.

**Comment by antquinonez** [comment]
@Altai-man , sounds good.

**Comment by WildYorkies** [comment]
@Altai-man Do you want someone to review it for grammar/spelling stuff? Or are you looking for factual correctness?

**Comment by antquinonez** [comment]
I think I got the grammar down, and mostly want to make sure the facts are
right/I didn't make a mistake. But it's all fair game.

Thanks.

**Comment by Altai-man** [comment]
@WildYorkies, if the meaning is the same, then we need to check, does that rewording _really_ made things better/more native or it is just rewording. If some pieces was changed greatly, we need to check it too, But examples wasn't changed, so it makes me think that it is only grammar/style that needs to be reviewed.

**Comment by WildYorkies** [comment]
@Altai-man Let me go through the tutorial today on @antquinonez 's branch. I've never done it. Can at least give my thoughts on clarity.

**Comment by Altai-man** [comment]
@WildYorkies, I'm not a boss here, so you can do as you think appropriate without my permission. I'm personally 👍 for any review of this.

File: doc/Language/grammar_tutorial.pod6 [hunk 1]
```diff
@@ -621,16 +559,14 @@ And this is the grammar and grammar actions that got us there, to recap:
     }
     =end code
```
Comment (WildYorkies) [inline doc/Language/grammar_tutorial.pod6]: We should also include an alternate version that splits the `data` into smaller pieces by just using the grammar methods. The whole time that was in my mind like "why do we need an action for this"? So perhaps we can quell questions by just including the alternate as a side note.

File: doc/Language/grammar_tutorial.pod6 [hunk 2]
```diff
@@ -4,98 +4,75 @@
...
+or to identify the different parts of a sentence, among other things.
 
 =SUBTITLE When Would I Use Grammars?
 
-Grammars are great at taking strings, trying to make sense of them, and then
-saving them into a data structure you can actually work with. If you have
-strings of some kind that you need to bring order to, or interpret, grammars
-give you some great tools to make it easier.
+If you have strings to tame or interpret, grammars provide the
+tools to do the job.
```
Comment (antquinonez) [inline doc/Language/grammar_tutorial.pod6]: I know. :) maybe an adjective phrase.

If you have overly complicated strings to tame or interpret, grammars provide the tools to do the job.

File: doc/Language/grammar_tutorial.pod6 [hunk 3]
```diff
@@ -4,98 +4,75 @@
 
 =SUBTITLE Why Grammars?
 
-You will always run into string parsing that will give you a headache. It is
-said that HTML, for example, cannot be broken down and parsed effectively at
-all, simply by using regular expressions to sort out the elements. Another
-example would be defining the order in which words and symbols might make up a
-language and provide meaning. This is when Perl's grammar system fits
-perfectly.
+Grammars parse strings and return data structures from those strings. Grammars can
```
Comment (antquinonez) [inline doc/Language/grammar_tutorial.pod6]: I can switch back.

File: doc/Language/grammar_tutorial.pod6 [hunk 4]
```diff
@@ -4,98 +4,75 @@
...
-other patterns - regular expressions alone are not adequate.
+Regular expressions work well for finding patterns in strings. However, for some
+tasks -- like finding multiple patterns at once, or
+combining patterns, or testing for patterns that may surround strings --
+regular expressions, alone, are not enough.
 
-Grammars provide a way to define how you want to examine a string using regular
-expressions, and you can group these regular expressions together to provide
-even more meaning.
+When working with HTML, you could define a grammar to recognize HTML tags, both
```
Comment (antquinonez) [inline doc/Language/grammar_tutorial.pod6]: I'll take a look at this.

File: doc/Language/grammar_tutorial.pod6 [hunk 5]
```diff
@@ -104,50 +81,45 @@ will be keyed with the same name as the method you chose to name it.
     }
```
Comment (antquinonez) [inline doc/Language/grammar_tutorial.pod6]: will do.

File: doc/Language/grammar_tutorial.pod6 [hunk 6]
```diff
@@ -104,50 +81,45 @@ will be keyed with the same name as the method you chose to name it.
...
     =end code
 
-So in this, if you were to my $match = My::Gram.parse( $string ) -- and your
-string started with the characters 'clever_text_keyword', you would get a match
+If you were to use my $match = My::Gram.parse( $string ) and your
+string started with 'clever_text_keyword', you would get a match
 object back that contained 'clever_text_keyword' keyed by the name of 'thingy'
-in your match object. These can grow much more complex, as your needs require,
-as you might imagine.
+in your match object.
```
Comment (antquinonez) [inline doc/Language/grammar_tutorial.pod6]: ok.

File: doc/Language/grammar_tutorial.pod6 [hunk 7]
```diff
@@ -104,50 +81,45 @@ will be keyed with the same name as the method you chose to name it.
...
 
-Now, to mention TOP. The TOP method (regex, token or rule) is the overarching
-regex that must match everything (by default). If the string you pass in to
-parse doesn't match the TOP regex, your returned match object will be empty
-(Any).
+The TOP method (whether regex, token, or rule) is the overarching pattern that must
+match everything (by default). If the parsed string doesn't match the TOP regex,
+your returned match object will be empty (Any).
```
Comment (antquinonez) [inline doc/Language/grammar_tutorial.pod6]: ok

File: doc/Language/grammar_tutorial.pod6 [hunk 8]
```diff
@@ -157,15 +129,15 @@ since we don't care about whitespace in the regex.
...
+be everything else left in the string.
 
-But in our big string we get, we don't know what order these regex matches will
-come in. We need to be able to place these matching tokens in the larger
-context of our URI we'll be passing in as that string. That's what the TOP
-method is for. So we add it, and place our tokens by name within it, along with
-however else our valid string should look, coming in.
+Next, we'll want to arrange these matching tokens within the larger context of
+the URI. That's what the TOP method allows us to do. We'll add the TOP method
+and place the names of our tokens within it, together with the rest of of the
```
Comment (antquinonez) [inline doc/Language/grammar_tutorial.pod6]: thanks.

File: doc/Language/grammar_tutorial.pod6 [hunk 9]
```diff
@@ -189,21 +160,22 @@ that has all 3 parameters included:
     #  data => ｢7/notify｣
```
Comment (antquinonez) [inline doc/Language/grammar_tutorial.pod6]: You see them in the console output when you run the say's was my recollection.
HALFWIDTH LEFT CORNER BRACKET is what a UNICODE search says. I can remove them. They were present in the original version.

File: doc/Language/grammar_tutorial.pod6 [hunk 10]
```diff
@@ -189,21 +160,22 @@ that has all 3 parameters included:
...
-The REST grammar so far will handle retrieves, deletes and updates ok. However,
+So far, the grammar will handle retrieves, deletes and updates. However,
 a I<create> command doesn't have the third part (the I<data> portion). This
-means our grammar will fail to match if we try to parse a create URL, and
-everyone will scream. To avoid this, we need to make that last I<data> position
-match optional, along with the '/' preceding it. This is easily accomplished by
-adding a question mark for the grouped '/' and I<data> components of the TOP
-token, to indicate their optional nature, just like a normal regex. So now we
-have:
+means the grammar will fail to match if we try to parse a create URI. To avoid
```
Comment (antquinonez) [inline doc/Language/grammar_tutorial.pod6]: ok.

File: doc/Language/grammar_tutorial.pod6 [hunk 11]
```diff
@@ -347,39 +315,27 @@ Let's look at various URIs and how they behave being passed through our grammar.
...
-class that you create, and use from your grammars to do stuff with the matches
-you find in your grammars.
+Grammar actions are used within grammar classes to do things with matches.
+Actions are defined in their own classes, distinct from grammar classes.
 
 You can think of grammar actions as a kind of plug-in expansion module for
-grammars. A lot of the time you'll be happy using grammars just on their own.
-But when you need to process some of those strings further, you can plug in the
-expansion module.
+grammars. A lot of the time you'll be happy using grammars all by their own.
```
Comment (antquinonez) [inline doc/Language/grammar_tutorial.pod6]: Thanks.

File: doc/Language/grammar_tutorial.pod6 (hunk 9 above)
Comment (WildYorkies) [inline doc/Language/grammar_tutorial.pod6]: If that's what it outputs in the console, then I'd keep it. I'm just wondering why they're there and why nothing else uses them. If they're just objects, why aren't they printed like other objects?

**Comment by Altai-man** [comment]
@tbrowder++ approved, hence merged. Thanks for the contribution!

**Pull request merged**